synapseclient/api/wiki_service.py

```diff
@@ -0,0 +1,456 @@
+<https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.WikiController2>
+"""
+
+import json
+from typing import TYPE_CHECKING, Any, AsyncGenerator, Dict, List, Optional
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+
+async def post_wiki(
+    owner_id: str,
+    request: Dict[str, Any],
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """Create a new wiki page.
+    <https://rest-docs.synapse.org/rest/POST/entity/ownerId/wiki2.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        request: The wiki page to create.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The created wiki page.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_post_async(
+        uri=f"/entity/{owner_id}/wiki2",
+        body=json.dumps(request),
+    )
+
+
+async def get_wiki_page(
+    owner_id: str,
+    *,
+    wiki_id: Optional[str] = None,
+    wiki_version: Optional[int] = None,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """Get a wiki page.
+    <https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki2.html>
+    <https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki2/wikiId.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        wiki_id: Optional ID of the wiki. If not provided, returns the root wiki page.
+        wiki_version: Optional version of the wiki page.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The requested wiki page.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Construct the URI based on whether wiki_id is provided
+    uri = f"/entity/{owner_id}/wiki2"
+    if wiki_id is not None:
+        uri = f"{uri}/{wiki_id}"
+
+    # Add version as a query parameter if provided
+    params = {}
+    if wiki_version is not None:
+        params["wikiVersion"] = wiki_version
+
+    return await client.rest_get_async(
+        uri=uri,
+        params=params,
+    )
+
+
+async def put_wiki_page(
+    owner_id: str,
+    wiki_id: str,
+    request: Dict[str, Any],
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """Update a wiki page.
+    <https://rest-docs.synapse.org/rest/PUT/entity/ownerId/wiki2/wikiId.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        wiki_id: The ID of the wiki.
+        request: The updated wiki page.
```

Comment:
Could there be a link here to the rest api doc page for the exact object/fields that `request` is supposed to be?

Reply:
Done